src/app/angular/cdk-tree/cdk-tree.component.ts

```diff
@@ -97,7 +97,13 @@ export class CdkTreeComponent {
   }
 
   shouldRender(node: ExampleFlatNode) {
-    const parent = this.getParentNode(node);
-    return !parent || parent.isExpanded;
+    let parent = this.getParentNode(node);
+    while (parent) {
+      if (!parent.isExpanded) {
+        return false;
+      }
+      parent = this.getParentNode(parent);
+    }
+    return true;
```

Comment:
This could be optionally simplified with:

```suggestion
    const parent = this.getParentNode(node);
    return !parent || (parent.isExpanded && this.shouldRender(parent));
```

(It's a recursive call, but given the depth we are dealing with, it is more than fine.)

Reply:
Great improvement!
Changed